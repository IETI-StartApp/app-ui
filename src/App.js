import React from "react";
import './App.css';
import {AuthProvider} from "./authServices/Auth";
import {
    LandingPage,
    LoginPage,
    ProjectDescription,
    ProjectsGridPage,
    RegisterForm,
    RegisterProjectPage,
    UserSelect
} from "./pages";
import {BrowserRouter as Router, Route, Switch} from "react-router-dom"
import {PrivateRoute} from "./components";

function App() {
    return (
        <>
            <Router>
                <AuthProvider>
                    <Switch>
                        <Route exact path="/" render={() => <LandingPage/>}/>
                        <Route exact path="/signin" render={() => <LoginPage/>}/>
                        <Route exact path='/signup' render={() => <RegisterForm/>}/>
                        <PrivateRoute exact path='/project-description' component={ProjectDescription}/>
                        <PrivateRoute exact path='/register-project' component={RegisterProjectPage}/>
                        <Route exact path='/user-type' component={UserSelect}/>
                        <Route exact path='/projects' component={ProjectsGridPage}/>
                    </Switch>
                </AuthProvider>
            </Router>
        </>
    );
}

export default App;
