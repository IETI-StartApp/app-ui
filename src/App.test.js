import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import App from "./App";


describe('Testing progress bar', () => {

    const wrapper = shallow(<App/>)

    test('should render appÂº', () => {
        expect(wrapper).toMatchSnapshot();
    })

})
