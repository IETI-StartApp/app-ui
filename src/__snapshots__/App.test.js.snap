// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing progress bar should render appÂº 1`] = `
<Fragment>
  <BrowserRouter>
    <AuthProvider>
      <Redirect
        exact={true}
        from="/"
        to="/login"
      />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/login"
        />
        <PrivateRoute
          component={[Function]}
          exact={true}
          path="/project-description"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/signup"
        />
        <PrivateRoute
          component={[Function]}
          exact={true}
          path="/register-project"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/user-type"
        />
      </Switch>
    </AuthProvider>
  </BrowserRouter>
</Fragment>
`;
